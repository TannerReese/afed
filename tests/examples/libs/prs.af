# Test prime number related functions

# Erroneous cases
prime_factors 4.32 = ``,
divisors [1, 2] = ``,
radical (-13) = ``,

# Prime number operations
prs.primes 47 = ``,
filter is_prime (range 0 47) = ``,
prime_factors 73827 = ``,

# Square free operations
is_sqfree r = ``, is_sqfree 60 = ``,
r: radical (2^4 * 3^3 * 5^7 * 101) = ``,

# Other operations
map euler_totient (range 0 20) = ``,
divisors 84 = ``,
sum $ (divisors x).map \d: d^2 = ``,
divisor_sum 2 (x: 67328) = ``,
filter prs.is_perfect (range 0 10000) = ``,

