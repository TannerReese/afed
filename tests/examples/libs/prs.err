line 4, column 22 Eval Error: Cannot cast number to integer type
line 5, column 19 Eval Error: Expected number, but found array
line 6, column 17 Eval Error: Cannot cast number to integer type
3 Eval Errors encountered
