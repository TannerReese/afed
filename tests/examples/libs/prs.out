# Test prime number related functions

# Erroneous cases
prime_factors 4.32 = `Eval Error: Cannot cast number to integer type`,
divisors [1, 2] = `Eval Error: Expected number, but found array`,
radical (-13) = `Eval Error: Cannot cast number to integer type`,

# Prime number operations
prs.primes 47 = `[2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47]`,
filter is_prime (range 0 47) = `[2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47]`,
prime_factors 73827 = `[[3, 2], [13, 1], [631, 1]]`,

# Square free operations
is_sqfree r = `true`, is_sqfree 60 = `false`,
r: radical (2^4 * 3^3 * 5^7 * 101) = `3030`,

# Other operations
map euler_totient (range 0 20) = `[1, 1, 1, 2, 2, 4, 2, 6, 4, 6, 4, 10, 4, 12, 6, 8, 8, 16, 6, 18, 8]`,
divisors 84 = `[1, 2, 4, 3, 6, 12, 7, 14, 28, 21, 42, 84]`,
sum $ (divisors x).map \d: d^2 = `6044143770`,
divisor_sum 2 (x: 67328) = `6044143770`,
filter prs.is_perfect (range 0 10000) = `[6, 28, 496, 8128]`,

